ably/error.go

```diff
@@ -135,3 +136,55 @@ func checkValidHTTPResponse(resp *http.Response) error {
 	}
 	return err
 }
+
+// ErrorInfoV12 is an error produced by the Ably library.
+type ErrorInfoV12 struct {
+	StatusCode int
+	Code       int
+	HRef       string
+	Message    string
+	Server     string
+}
+
+// Error implements the builtin error interface.
+func (e ErrorInfoV12) Error() string {
+	errorHref := e.HRef
+	if errorHref == "" && e.Code != 0 {
+		errorHref = fmt.Sprintf("https://help.ably.io/error/%d", e.Code)
+	}
+	var see string
+	if !strings.Contains(e.Message, errorHref) {
+		see = "See " + errorHref
+	}
+	return fmt.Sprintf("[ErrorInfo :%s code=%d statusCode=%d] %s", e.Message, e.Code, e.StatusCode, see)
+}
```

Comment:
I don't see a reason to put ErrorInfo in a separate, internal-ish package. Per the spec, it's a public type. Ultimately, I guess one of them will go away, but I'm focusing first on getting the spec-compliant API out and then see how we can refactor things to deduplicate.

Reply:
Hmm ok, well should we at least add a `TODO: remove duplication with proto.ErrorInfo` comment?

I must admit though that I don't like the idea of creating more work for ourselves down the line rather than just doing the work now when we know it needs to be done (e.g. rather than copying and pasting code around, just move or share it).

Reply:
Done at d8fbcf8.

Reply:
More changes related to this at 2eccadbf7c0835f74040af7264d6820c736b885f.